### PR TITLE
Add integration test for diff --prune --selector

### PR DESCRIPTION
**Pull request by ardaguclu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR adds new integration tests for `kubectl diff --prune -l` to catch possible regressions in the future.

issue: https://github.com/kubernetes/kubectl/issues/1355
PR fixing the issue: https://github.com/kubernetes/kubernetes/pull/114863

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by ardaguclu** [comment]
/hold
for review.

**Comment by ardaguclu** [comment]
First commit of this PR disables label selector just like before the fix. Thereby, integration test should fail. After integration test fails, second commit will enable label selector again and expects integration test should pass.

/cc @KnVerey @nathanmartins

**Comment by ardaguclu** [comment]
It failed as expected https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/115571/pull-kubernetes-integration/1622870030662242304

File: test/cmd/diff.sh [hunk 1]
```diff
@@ -113,6 +113,89 @@ run_kubectl_diff_tests() {
...
+    prune-group: "false"
+spec:
+  containers:
+    - name: kubernetes-pause
+      image: registry.k8s.io/pause:3.9
+EOF
+    kube::test::get_object_assert 'pods a -n nsbprune' "{{${id_field:?}}}" 'a'
+    kube::test::get_object_assert 'pods b -n nsbprune' "{{${id_field:?}}}" 'b'
+    kube::test::get_object_assert 'pods c -n nsbprune' "{{${id_field:?}}}" 'c'
+    # Make sure that kubectl diff does not return neither pod 'b' nor pod 'c' without prune flag
```
Comment (mpuckett159) [inline test/cmd/diff.sh:160]:
```suggestion
    # Make sure that kubectl diff does not return either pod 'b' or pod 'c' without prune flag
```
I think this is more clear.

File: test/cmd/diff.sh [hunk 2]
```diff
@@ -113,6 +113,89 @@ run_kubectl_diff_tests() {
...
+spec:
+  containers:
+    - name: kubernetes-pause
+      image: registry.k8s.io/pause:3.9
+EOF
+)
+    output_message=$(echo "${PRUNE}" | kubectl diff -l prune-group=true -f -)
+    kube::test::if_has_not_string "${output_message}" "name: b"
+    kube::test::if_has_not_string "${output_message}" "name: c"
+    # the exist code for diff is 1 because pod name 'b' is found in the given label selector but not 'c'
```
Comment (mpuckett159) [inline test/cmd/diff.sh:190]:
```suggestion
    # the exit code for diff is 1 because pod 'b' is found in the given label selector but not 'c'
```

**Review by mpuckett159 — COMMENTED** [review]
Just some minor corrections to the comments, otherwise lgtm.

**Comment by ardaguclu** [comment]
@mpuckett159 Thanks for review, I applied your suggested changes. Could you PTAL one last round?

**Pull request merged**